docs/hw/graph/maxwell/cuda/int.rst

```diff
@@ -0,0 +1,54 @@
+.. _maxwell-int:
+
+===============================
+Integer Arithmetic Instructions
+===============================
+
+.. contents::
+
+Introduction
+============
+
+Common Flags
+============
+
+neg
+---
+
+Negate the operand.
+
+h0/h1
+-----
+
+An optional flag that can be either H0 or H1. With H1, the high 16 bits of the operand are used. With H0, the low 16 bits.
+
+x
+-
+
+Use the carry flag.
+
+cc
+--
+
+Set the carry flag.
+
+.. _maxwell-opg-iadd3
+
+Addition: iadd3
+===============
+
+::
+
+  iadd3 [mode,x,cc] REG0 [neg,h0/h1] REG1 [neg,h0/h1] REG2 [neg,h0/h1] REG3
+  iadd3 [mode,x,cc] REG0 [neg,h0/h1] REG1 [neg,h0/h1] REG2 [neg,h0/h1] CB3
```

Comment:
Are you sure that the h0/h1 forms exist for the latter 2 versions?
Also, usually the const/imm arg is second, not third - are you sure you got this right?
And lastly, you missed the limm form (in such forms, usually dstreg == src3reg)